### ISSUE TITLE
Error in coala tutorial. Coala failed to apply automatic patch

### ISSUE THREAD

**Issue by shiv12095** [coala-analyzer/coala#1060]
```
/home/shiv/coala/coala-tutorial/src/main.c
|   6|     
|    | [NORMAL] SpaceConsistencyBear:
|    | Line contains following spacing inconsistencies:
|    | - Trailing whitespaces.
|    | The following actions are applicable to this result:
|    |  0: Apply no further actions.
|    |  1: Open the affected file(s) in an editor.
|    |  2: Apply the patch automatically.
|    |  3: Print a diff of the patch that would be applied.
|    | Please enter the number of the action you want to execute. 2
|    | The following actions are applicable to this result:
|    |  0: Apply no further actions.
|    |  1: Open the affected file(s) in an editor.
|    |  2: Apply the patch automatically.
|    |  3: Print a diff of the patch that would be applied.
|    | Please enter the number of the action you want to execute. 2
[ERROR][00:47:31] Failed to execute the action ApplyPatchAction.
|    | The following actions are applicable to this result:
|    |  0: Apply no further actions.
|    |  1: Open the affected file(s) in an editor.
|    |  2: Apply the patch automatically.
|    |  3: Print a diff of the patch that would be applied.
|    | Please enter the number of the action you want to execute. 2
[ERROR][00:47:37] Failed to execute the action ApplyPatchAction.
|    | The following actions are applicable to this result:
|    |  0: Apply no further actions.
|    |  1: Open the affected file(s) in an editor.
|    |  2: Apply the patch automatically.
|    |  3: Print a diff of the patch that would be applied.
|    | Please enter the number of the action you want to execute. 2

```

**Comment by Makman2** [comment]
confirmed
coala asks again what action to perform after applying diff. The diff is applied and so not valid any more.

**Comment by Makman2** [comment]
@shiv12095 thanks for reporting this bug :)

**Comment by sils1297** [comment]
Hey,

Thanks for reporting this! Any feedback or bug reports, especially related
to newcomer material is greatly appreciated.

This functionality is kind of intended. We have a very flexible action
system, so some actions might make sense to apply repeatedly. What I
conclude how we could improve here is:

A) introduce a property of actions that are only applicable once, remove
them from the choice once applied.
B) show a confirmation that the action actually got executed successfully.
Guessing from your log coala corrected the issue automatically the first
time you entered the 2.
C) improve the getting started tutorial concerning this.

Would that suit your needs?

**Comment by sils1297** [comment]
@shiv12095 can you please also confirm that/if the patch actually got applied?

**Comment by shiv12095** [comment]
@sils1297 

A combination of points A and B would be more intuitive and helpful. 

So you could give a confirmation message saying "Patch <Patch Name> has been successfully applied" Since some actions can be applied multiple times, you should continue giving that but actions that are no longer available should be removed from the list of possible actions.

The patch worked fine

**Comment by sils1297** [comment]
@shiv12095 cool! I've added a success message for now at https://github.com/coala-analyzer/coala/pull/1063 , more will follow. The other thing is a bit nifty to do right I think.

**Comment by shiv12095** [comment]
The confirmation message should suffice for now. The other feature is not urgent at the moment

**Comment by sils1297** [comment]
@shiv12095 it's almost ready anyway, just need to write the tests and you'll get it

**Comment by sils1297** [comment]
ok takes a little longer but we'll get to it within a few days I guess.

**Comment by sils1297** [comment]
This bug will be marked closed when aaca9a5 gets merged. I filed https://github.com/coala-analyzer/coala/pull/1063 for the applicability of actions reevaluation. I wrote https://github.com/coala-analyzer/coala/commit/80c2d5529b96ed747062746cc129849bc111713f which is a bit more sophisticated then just checking if an action was already applied: it checks if there will be conflicts before offering the option. This way you could e.g. use the OPenEditorAction to change the file manually and *after* that the patch will be appliable if, and only if, it applies cleanly without conflicts.

**Comment by sils1297** [comment]
However in any case the user will still be able to *show* the patch.
